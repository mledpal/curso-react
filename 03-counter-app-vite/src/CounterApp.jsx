import { useState } from  'react';
import PropTypes from 'prop-types';

export const CounterApp = ( { value } ) => {

    const [ counter, setCounter ] = useState( value );

    const sumarContador = () => setCounter( counter + 1 );
    const restarContador = () => setCounter( counter - 1 );
    const resetContador = () => setCounter(value);

    return (
        <>
            <h1>CounterApp</h1>
            <h2>{ counter } </h2>
            <button onClick={ sumarContador }> +1 </button>
            <button onClick={ restarContador }> -1 </button>
            <button onClick={ resetContador }> Reset </button>
        </>
    )
}

CounterApp.propTypes = {
    value: PropTypes.number.isRequired,
}

CounterApp.defaultProps = {
    value: 0,
}