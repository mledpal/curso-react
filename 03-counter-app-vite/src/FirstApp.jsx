import PropTypes from 'prop-types';

export const FirstApp = ( { titulo, subtitulo, nombre } ) => {

    // if(!titulo) {
    //     throw new Error('El título no existe');
    // }

    return (
        <>
        {/* <code>{ JSON.stringify (newMessage) }</code> */}
        <h1 data-testid="test-title">{ titulo }</h1>                
        <p>{ subtitulo }</p>
        <p>{ subtitulo }</p>
        <p>{ nombre }</p>
        </>
    );
}

FirstApp.propTypes = {
    nombre: PropTypes.string,
    subtitulo: PropTypes.string,
    titulo: PropTypes.string.isRequired,
}

FirstApp.defaultProps = {
    nombre: 'Miguel Ledesma',
    subtitulo: 'Subtitulo',
    titulo : 'No hay título',
}