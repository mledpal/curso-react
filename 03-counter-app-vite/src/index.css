* {
    margin: 0;
    padding: 0;
    box-sizing: border-box;    
}

body {
    align-items: center;
    background-image: linear-gradient(90deg, #0d4da0, #4f96cf);
    color: #eee;
    display: flex;
    flex-direction: column;
    font-family: System-ui;
    font-weight: 600;
    justify-content: center;
    min-height: 100vh;
    padding: 20px;
    width: 100%;
    text-align: center;
}

h1 {
    font-size: 2em;
}

button {
    padding: 10px 30px;
    border-radius: 5px;
    box-shadow: 1px 1px 4px #222, -1px -1px 4px #555 inset;
    font-size: 1.3rem;
    margin: 5px;
}