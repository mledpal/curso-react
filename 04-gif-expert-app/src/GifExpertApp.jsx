import { useState } from "react";
import { AddCategory, GiftGrid } from './components';


export const GifExpertApp = () => { 

    const [categories, setCategories] = useState(['Dragon Ball']);

    const onAddCategory = ( nuevaCategoria ) => {
        
        if(!categories.includes(nuevaCategoria)){
            setCategories( [nuevaCategoria, ...categories] );
        } else {
            console.log('Ya existe');
        }
        
    };

    return (
        <>
            
            <h1>GifExpertApp</h1>

            <AddCategory onAddCategory = { onAddCategory }/>
            
            { categories.map( category => ( 
                <GiftGrid 
                    key={ category } 
                    category={ category } 
                /> 
            ))}
        </>
    )

}