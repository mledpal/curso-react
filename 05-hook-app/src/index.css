* { 
  margin: 0;
  padding: 0;
  box-sizing: border-box;
}

body {
  width: 100vw;
  max-width: 100%;
  min-height: 100vh;
  font-family: system-ui;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
}

button {  
  padding: 20px;
  margin: 10px;
}