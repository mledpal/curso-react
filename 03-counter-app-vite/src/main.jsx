/* eslint-disable react-refresh/only-export-components */
/* eslint-disable no-unused-vars */

import React from 'react';
import ReactDOM from 'react-dom/client';

// import { FirstApp } from './FirstApp';
// import { App } from './HelloWorldApp';
import { CounterApp } from './CounterApp';

import './index.css';

const saludo = {
    titulo : 'Este es el titulo',
    subtitulo : 'Este es el subtitulo',
    nombre : 'Miguel Ledesma'
}

ReactDOM.createRoot(document.getElementById("root")).render(
    <React.StrictMode>        
        {/* <FirstApp titulo = {saludo.titulo} subtitulo = {saludo.subtitulo} nombre = {saludo.nombre} /> */}
        <CounterApp value = { 0 }/> 
    </React.StrictMode>
)