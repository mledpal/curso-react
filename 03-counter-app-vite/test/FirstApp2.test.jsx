/* eslint-disable no-unused-vars */
/* eslint-disable no-undef */

import { getAllByText, getByText, render, screen } from '@testing-library/react';
import { FirstApp } from '../src/FirstApp';

describe('Pruebas con FirstApp', () => { 

    const titulo    = "Hola, soy Miguel";
    const subtitulo = "Hola, esto es un subtitulo";


    test('Debe hacer match con snapshot', () => {            
       const { container } = render(<FirstApp titulo={ titulo } />);
       expect(container).toMatchSnapshot();        
    });


    test('Debe mostrar el mensaje "Hola, soy Miguel" en un p', () => { 
        render(<FirstApp titulo={ titulo } />);
        expect(screen.getByText(titulo)).toBeTruthy();
    });


    test('Debe encontrar el titulo en un H1', () => { 
        render(<FirstApp titulo={ titulo } />);
        expect(screen.getByRole('heading', { level: 1 }).innerHTML).toContain(titulo);
    });


    test('Debe mostrar el subtitulo enviado por props', () => { 
        
        render( 
            <FirstApp 
                titulo={ titulo } 
                subtitulo = { subtitulo }
            /> 
        );

        expect(screen.getAllByText(subtitulo).length).toBe(2);
    });
});