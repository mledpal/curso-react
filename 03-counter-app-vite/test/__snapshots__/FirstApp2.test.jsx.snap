// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas con FirstApp Debe hacer match con snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hola, soy Miguel
  </h1>
  <p>
    Subtitulo
  </p>
  <p>
    Subtitulo
  </p>
  <p>
    Miguel Ledesma
  </p>
</div>
`;
